server/src/main/java/org/elasticsearch/action/admin/cluster/node/reload/NodesReloadSecureSettingsRequestBuilder.java

```diff
@@ -19,22 +19,66 @@
+import org.elasticsearch.common.xcontent.NamedXContentRegistry;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.common.xcontent.XContentType;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Objects;
 
 /**
  * Builder for the reload secure settings nodes request
  */
 public class NodesReloadSecureSettingsRequestBuilder extends NodesOperationRequestBuilder<NodesReloadSecureSettingsRequest,
         NodesReloadSecureSettingsResponse, NodesReloadSecureSettingsRequestBuilder> {
 
+    public static final String SECURE_SETTINGS_PASSWORD_FIELD_NAME = "secure_settings_password";
+
     public NodesReloadSecureSettingsRequestBuilder(ElasticsearchClient client, NodesReloadSecureSettingsAction action) {
         super(client, action, new NodesReloadSecureSettingsRequest());
     }
 
-    public NodesReloadSecureSettingsRequestBuilder setSecureStorePassword(String secureStorePassword) {
+    public NodesReloadSecureSettingsRequestBuilder setSecureStorePassword(SecureString secureStorePassword) {
         request.secureStorePassword(secureStorePassword);
         return this;
     }
 
+    public NodesReloadSecureSettingsRequestBuilder source(BytesReference source, XContentType xContentType) throws IOException {
+        Objects.requireNonNull(xContentType);
+        // EMPTY is ok here because we never call namedObject
+        try (final InputStream stream = source.streamInput();
+                final XContentParser parser = xContentType.xContent().createParser(NamedXContentRegistry.EMPTY,
+                        LoggingDeprecationHandler.INSTANCE, stream)) {
+            XContentParser.Token token;
+            token = parser.nextToken();
+            if (token != XContentParser.Token.START_OBJECT) {
+                throw new ElasticsearchParseException("expected an object, but found token [{}]", token);
+            }
+            token = parser.nextToken();
+            if (token != XContentParser.Token.FIELD_NAME || false == SECURE_SETTINGS_PASSWORD_FIELD_NAME.equals(parser.currentName())) {
+                throw new ElasticsearchParseException("expected a field named [{}], but found [{}]", SECURE_SETTINGS_PASSWORD_FIELD_NAME,
+                        token);
+            }
+            token = parser.nextToken();
+            if (token != XContentParser.Token.VALUE_STRING) {
+                throw new ElasticsearchParseException("expected field [{}] to be of type string, but found [{}] instead",
+                        SECURE_SETTINGS_PASSWORD_FIELD_NAME, token);
+            }
+            final String password = parser.text();
+            setSecureStorePassword(new SecureString(password.toCharArray()));
+            token = parser.nextToken();
+            if (token != XContentParser.Token.END_OBJECT) {
+                throw new ElasticsearchParseException("expected end of object, but found token [{}]", token);
+            }
+        }
+        return this;
+    }
+
```

Comment:
1/2: Since right now the keystore only uses the empty password, the previous implementation was sloppy and actually read the password as a REST parameter. This is terrible and has been corrected. It is now read from the request body.

Reply:
We're going to need to figure out a way to filter this in the security auditing. In x-pack we introduced the concept of a `RestRequestFilter`. This concept might need to be promoted to core for this API.